frontend/src/lib/keybind.ts

```diff
@@ -211,36 +203,37 @@ const openResultInNewTab = (e: KeyboardEvent) => {
  *
  * @param e - The triggering {@link KeyboardEvent}
  */
-const domainSearch = (e: KeyboardEvent) => {
-  // Ensure target event is a result url
-  if ((e.target as HTMLElement).className.includes('result-main-link')) {
-    const searchBar = document.getElementById('searchbar') as HTMLInputElement;
-    if (searchBar) {
-      // Parse the result url, pull out the hostname then resubmit search
-      const resultLink = new URL((e.target as HTMLAnchorElement)?.href);
-      searchBar.value += ` site:${resultLink.hostname}`;
-      searchBar.form?.submit();
+const domainSearch = (e: KeyboardEvent, context: Refs) => {
+  if (e.target && e.target instanceof HTMLElement) {
+    const activeElement = e.target as HTMLElement;
+    const currentResultIndex = context.results.findIndex((el) =>
+      el.getMainDiv().contains(activeElement),
+    );
+    if (currentResultIndex > -1) {
+      const focusedResult = context.results[currentResultIndex];
+      const query = context.searchbar.userQuery();
+      const domain = focusedResult.getMainResultLink().getUrl().hostname;
+      const domainQuery = `site:${domain}`;
+
+      // Only run the domain query if it isn't already in the query
+      if (!query.includes(domainQuery)) context.searchbar.search(`${query} ${domainQuery}`);
     }
   }
 };
 
-/**
- * Redirect to the `Did you mean:` link (if exists)
- */
-const goToMisspellLink = () => {
```

Comment:
This is also the only one that I can't get working. I'm pretty sure if has to do with the logic of the spell correction and it's conditional rendering - let me know if you look over this and see anything right off the bat